plasmapy/diagnostics/thomson.py

```diff
@@ -0,0 +1,257 @@
+"""
+Defines the Thomson scattering analysis module as
+part of the diagnostics package.
+"""
+
+__all__ = [
+    "spectral_density",
+]
+
+import astropy.constants as const
+import astropy.units as u
+import numpy as np
+
+from plasmapy.formulary.parameters import plasma_frequency, thermal_speed
+from plasmapy.formulary.dielectric import permittivity_1D_Maxwellian
+from plasmapy.particles import Particle
+from plasmapy.utils.decorators import validate_quantities
+from typing import List, Tuple, Union
+from warnings import warn
+
+
+# TODO: interface for inputting a multi-species configuration could be
+# simplified using the plasmapy.classes.plasma_base class if that class
+# included ion and electron drift velocities and information about the ion
+# atomic species.
```

Comment:
This is the kind of comment I like to see!

I think we should have a general plasma class design issue to track this sort of thing. Note to self...